### PR TITLE
Replace cygwin github action

### PR DESCRIPTION
**Pull request by piponazo**
This PR should finally fix the cygwin build

### CODE REVIEW

File: .github/workflows/on_PR_windows_matrix.yml [hunk 1]
```diff
@@ -156,15 +156,15 @@ jobs:
...
+        shell: C:\cygwin\bin\bash.exe -eo pipefail '{0}'
     steps:
       # Make sure we don't check out scripts using Windows CRLF line endings
       - run: git config --global core.autocrlf input
         shell: pwsh
       - uses: actions/checkout@v3
 
       - name: Set up Cygwin
-        uses: egor-tensin/setup-cygwin@v3
+        uses: cygwin/cygwin-install-action@master
```
Comment (kmilos) [inline .github/workflows/on_PR_windows_matrix.yml:167]: Should we maybe go for a known tag, v3 currently?

**Comment by kmilos** [comment]
Nice, it also seems to resolve package dependencies as well!

**Review by kevinbackhouse — DISMISSED** [review]
Awesome! Thanks for figuring this out!

File: .github/workflows/on_PR_windows_matrix.yml (hunk 1 above)
Comment (piponazo) [reply .github/workflows/on_PR_windows_matrix.yml:167]: Yes, I will make the change :)

**Review by kmilos — APPROVED** [review]

**Pull request merged**